conf/modules.config

```diff
@@ -263,80 +272,159 @@ process {
             ]
     }
 
+    withName: FILTER_BAM_FRAGMENT_LENGTH {
+        tag = { "${meta.reference}|${meta.sample_id}_${meta.library_id}" }
+        ext.args = "-l ${params.bamfiltering_minreadlength}"
+        ext.prefix = { "${meta.sample_id}_${meta.library_id}_${meta.reference}" }
+        publishDir = [
+            path: { "${params.outdir}/bam_filtering/" },
+            mode: params.publish_dir_mode,
+            pattern: '*.filtered.bam',
+            enabled: params.bamfiltering_savefilteredbams
+        ]
+    }
+
+    withName: SAMTOOLS_FASTQ_UNMAPPED {
+        tag = { "${meta.reference}|${meta.sample_id}_${meta.library_id}" }
+        ext.args = [
+            '-f 4',
+        ].join(' ').trim()
+        ext.prefix = { "${meta.sample_id}_${meta.library_id}_${meta.reference}_unmapped" }
+        publishDir = [
+            path: { "${params.outdir}/bam_filtering/" },
+            mode: params.publish_dir_mode,
+            pattern: '*.fastq.gz',
+            enabled: params.bamfiltering_generateunmappedfastq
+        ]
+    }
+
+    withName: SAMTOOLS_VIEW_BAM_FILTERING {
+        tag = { "${meta.reference}|${meta.sample_id}_${meta.library_id}" }
+        ext.args = [
+            "-q ${params.bamfiltering_mappingquality}",
+            params.bamfiltering_retainunmappedgenomicbam ? '' : "-F ${params.bamfilter_genomicbamfilterflag}",
+        ].join(' ').trim()
+        ext.prefix = { "${meta.sample_id}_${meta.library_id}_${meta.reference}_filtered" }
+        publishDir = [
+            path: { "${params.outdir}/bam_filtering/" },
+            mode: params.publish_dir_mode,
+            pattern: '*.bam',
+            enabled: params.bamfiltering_savefilteredbams
+        ]
+    }
+
+    withName: 'SAMTOOLS_LENGTH_FILTER_INDEX|SAMTOOLS_FILTER_INDEX' {
+        tag = { "${meta.reference}|${meta.sample_id}_${meta.library_id}" }
+        ext.args = { params.fasta_largeref ? "-c" : "" }
+        ext.prefix = { "${meta.sample_id}_${meta.library_id}_${meta.reference}_filtered" }
+        publishDir = [
+            path: { "${params.outdir}/bam_filtering/" },
+            mode: params.publish_dir_mode,
+            pattern: '*.{bai,csi}'
+        ]
+    }
+
+    withName: SAMTOOLS_FASTQ_MAPPED {
+        tag = { "${meta.reference}|${meta.sample_id}_${meta.library_id}" }
+        ext.args = [
+            params.metagenomicscreening_input == 'all' ? '' : '-F 4',
+        ].join(' ').trim()
+        ext.prefix = { "${meta.sample_id}_${meta.library_id}_${meta.reference}_mapped" }
+        publishDir = [
+            path: { "${params.outdir}/bam_filtering/" },
+            mode: params.publish_dir_mode,
+            pattern: '*.fastq.gz',
+            enabled: params.bamfiltering_generatemappedfastq
+        ]
+    }
+
+    withName: SAMTOOLS_FLAGSTAT_FILTERED {
+        tag = { "${meta.reference}|${meta.sample_id}_${meta.library_id}" }
+        ext.prefix = { "${meta.sample_id}_${meta.library_id}_${meta.reference}" }
```

Comment:
We should add something in the ext.prefix that indicates that this flagstats is filtered